pcapng/utils.py

```diff
@@ -69,3 +87,23 @@ def pack_timestamp_resolution(base, exponent):
     if base == 10:
         return struct.pack("B", exponent)
     raise ValueError("Supported bases are: 2, 10")
+
+
+# version-portable namedtuple with defaults, adapted from
+# https://stackoverflow.com/a/18348004/6692652
```

Comment:
Yeah I don't think we can use that. StackOverflow code snippets are licensed under a (date-dependent version of) CC-BY-SA license, which is not compatible with the Apache-2.0 license the module is licensed as.

I would say, if we need namedtuples with defaults, it's time to use https://www.attrs.org/en/stable/ instead.

Reply:
I didn't use any of the code in that answer, that's just where I got the idea. Take a look at the code in that answer for yourself and see if you think the code I've written is close enough to be worrisome. If it is, I'll look into `attrs`, otherwise I'll break this out into a new `_compat.py` as you recommended above.

Reply:
(And I'll remove the reference to stackoverflow to prevent future worries.)

Reply:
That's a bit better, although I would probably still wonder how much compat we want to have in general.

I'd remove the nested try/except for older Python 2.x, since even before dropping Python 2 support we only supported 2.7 anyway.